core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/handler/BaseAsyncClientHandler.java

```diff
@@ -60,44 +67,88 @@ protected BaseAsyncClientHandler(SdkClientConfiguration clientConfiguration,
     @Override
     public <InputT extends SdkRequest, OutputT extends SdkResponse> CompletableFuture<OutputT> execute(
         ClientExecutionParams<InputT, OutputT> executionParams) {
-        ExecutionContext executionContext = createExecutionContext(executionParams);
 
-        HttpResponseHandler<OutputT> decoratedResponseHandlers =
-            decorateResponseHandlers(executionParams.getResponseHandler(), executionContext);
+        validateExecutionParams(executionParams);
+        ExecutionContext executionContext = createExecutionContext(executionParams, createInitialExecutionAttributes());
+        TransformingAsyncResponseHandler<Response<OutputT>> combinedResponseHandler;
+
+        if (executionParams.getCombinedResponseHandler() == null) {
```

Comment:
Can we use some other predicates here to imply `All other protocols take this path` and ` // XML non-streaming protocol takes this path` in the `else` block?

It's not very clear to me that `executionParams.getCombinedResponseHandler() != null` means `XML non-streaming protocol`

Reply:
I don't think the comment is adding value. I'll try and break this code up into better named methods.